File: VB/Models/Invoice.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports GridViewLocalizeColumnCaptionMvc.Views.Home.App_LocalResources

Public Class Invoice
	Public Property Id() As Integer
	Public Property Description() As String
	Public Property Price() As Decimal

	<Display(Name := "Date", ResourceType := GetType(Localization))>
	Public Property RegisterDate() As Date

	Public Shared Function GetData() As List(Of Invoice)
		Dim invoices As New List(Of Invoice)()
		Const count As Integer = 3

		For i As Integer = 0 To count - 1
			invoices.Add(New Invoice() With {.Id = i, .Description = "Invoice" & i.ToString(), .Price = i * 10, .RegisterDate = Date.Today.AddDays(i - count)})
		Next i

		Return invoices
	End Function
End Class

File: CS/Models/Invoice.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using GridViewLocalizeColumnCaptionMvc.Views.Home.App_LocalResources;

public class Invoice {
    public int Id { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }

    [Display(Name = "Date", ResourceType = typeof(Localization))]
    public DateTime RegisterDate { get; set; }

    public static List<Invoice> GetData() {
        List<Invoice> invoices = new List<Invoice>();
        const int count = 3;

        for (int i = 0; i < count; i++) {
            invoices.Add(new Invoice() {
                Id = i,
                Description = "Invoice" + i.ToString(),
                Price = i * 10,
                RegisterDate = DateTime.Today.AddDays(i - count)
            });
        }

        return invoices;
    }
}